### PR TITLE
[circle-mpqsolver] Introduce H5FileDataProvider

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit introduces H5FileDataProvider as a wrapper over dio::hdf5::HDF5Importer.

Draft: #11849
Related: #11374
ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

File: compiler/circle-mpqsolver/src/core/DataProvider.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+void H5FileDataProvider::getSampleInput(uint32_t sample, uint32_t input, InputData &data) const
+{
+  if (_is_raw_data)
+  {
+    _importer.readTensor(sample, input, data._data.data(), data._data.size());
+  }
+  else
+  {
+    loco::DataType dtype;
+    Shape shape;
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/core/DataProvider.cpp:45]: These are not used. If you're going to define these variable, please use them to check dtype and shape of model input.

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/DataProvider.cpp:45]: > These are not used. If you're going to define these variable, please use them to check dtype and shape of model input.

ok. I need to update the draft at first.

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/DataProvider.cpp:82]: done

File: compiler/circle-mpqsolver/src/core/DataProvider.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+{
+namespace core
+{
+
+struct InputData
+{
+  std::vector<char> _data;
+
+  InputData(size_t size) : _data(size) {}
+};
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/core/DataProvider.h:37]: `_` prefix is often used for a private member. Can you use just a vector (remove this struct) or make the member private?

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/DataProvider.h:37]: > `_` prefix is often used for a private member. Can you use just a vector (remove this struct) or make the member private?

Ok.

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/DataProvider.h]: done

File: compiler/circle-mpqsolver/src/core/DataProvider.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  H5FileDataProvider(const std::string &h5file, const std::string &module_path);
+  size_t numSamples() const override;
+  uint32_t numInputs(uint32_t sample) const override;
+  void getSampleInput(uint32_t sample, uint32_t input, InputData &data) const override;
+
+private:
+  std::vector<loco::Node *> _input_nodes;
+  std::unique_ptr<luci::Module> _module;
+  dio::hdf5::HDF5Importer _importer;
+  bool _is_raw_data;
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/DataProvider.h:65]:
```suggestion
  bool _is_raw_data = false;
```
not sure our code analysis tool will complain about this...

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/DataProvider.h]: done

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**